consensus/src/chained_bft/block_storage/block_store.rs

```diff
@@ -398,10 +398,19 @@ impl<T: Payload> BlockStore<T> {
             .get_block(block.parent_id())
             .ok_or_else(|| format_err!("Block with missing parent {}", block.parent_id()))?;
         ensure!(parent.round() < block.round(), "Block with invalid round");
-        ensure!(
-            block.timestamp_usecs() > parent.timestamp_usecs(),
-            "Block with non-increasing timestamp"
-        );
+        if block.is_nil_block()
```

Comment:
similar comment as below, it seems like if we get into this position we likely have some weird error